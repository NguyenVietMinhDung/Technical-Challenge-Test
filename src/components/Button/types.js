export type Props = {
  btnTxt: string,
  handleClickEvent: Function,
};
