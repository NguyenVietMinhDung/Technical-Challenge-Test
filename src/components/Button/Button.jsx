import React from 'react';
import CSSModules from 'react-css-modules';
import styles from './Button.scss';
import type { Props } from './types';

const Button = (props: Props) => {
  const { handleClickEvent, btnTxt } = props;
  return (
    <button
      type="button"
      styleName="btn"
      onClick={handleClickEvent}
    >
      {btnTxt}
    </button>
  );
};

export default CSSModules(Button, styles);
