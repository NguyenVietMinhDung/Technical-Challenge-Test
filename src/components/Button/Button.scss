@import '../../styles/variables';

.btn {
  cursor: pointer;
  border: 0;
  color: $white-color;
  background-color: $violet-color;
  padding: 10px 20px;

  &:hover {
    border: 2px solid $violet-color;
    color: $violet-color;
    background-color: $white-color;
  }
}