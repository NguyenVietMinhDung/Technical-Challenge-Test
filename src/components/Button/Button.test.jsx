import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

const defaultProps = {
  btnTxt: 'LOAD MORE',
  handleClickEvent: jest.fn(),
};

describe('<Button />', () => {
  test('renders correctly', () => {
    const component = shallow(<Button {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
