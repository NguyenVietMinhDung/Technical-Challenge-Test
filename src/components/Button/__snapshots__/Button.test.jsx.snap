// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="btn"
  onClick={[MockFunction]}
  type="button"
>
  LOAD MORE
</button>
`;
